logdetective/server/database/models/merge_request_jobs.py

```diff
@@ -0,0 +1,473 @@
+          mr_iid: merge request forge iid
+          job_id: forge job id
+        """
+        mr = GitlabMergeRequestJobs.get_by_details(forge, project_id, mr_iid, job_id)
+        if mr is None:
+            id_ = GitlabMergeRequestJobs.create(forge, project_id, mr_iid, job_id)
+            mr = GitlabMergeRequestJobs.get_by_id(id_)
+        return mr
+
+
+class Comments(Base):
+    """Store details for comments
+    created by logdetective for a merge request job."""
+
+    __tablename__ = "comments"
+
+    id = Column(Integer, primary_key=True)
+    merge_request_job_id = Column(
+        Integer,
+        ForeignKey("gitlab_merge_request_jobs.id"),
+        nullable=False,
+        unique=True,  # 1 comment for 1 job
+        index=True,
+        comment="The associated merge request job (db) id",
+    )
+    comment_id = Column(
+        String(50),  # e.g. 'd5a3ff139356ce33e37e73add446f16869741b50'
+        nullable=False,
+        index=True,
+        comment="The comment gitlab id",
+    )
+    created_at = Column(
+        DateTime, nullable=False, comment="Timestamp when the comment was created"
+    )
+
+    merge_request_job = relationship("GitlabMergeRequestJobs", back_populates="comment")
+    reactions = relationship("Reactions", back_populates="comment")
+
+    @classmethod
+    @backoff.on_exception(backoff.expo, OperationalError, max_tries=DB_MAX_RETRIES)
+    def create(  # pylint: disable=too-many-arguments disable=too-many-positional-arguments
+        cls,
+        forge: Forge,
+        project_id: int,
+        mr_iid: int,
+        job_id: int,
+        comment_id: str,
+    ) -> int:
+        """Create a new comment id entry,
+        returns its PostgreSQL id.
+
+        Args:
+          forge: forge name
+          project_id: forge project id
+          mr_iid: merge request forge iid
+          job_id: forge job id
+          comment_id: forge comment id
+        """
+        mr_job = GitlabMergeRequestJobs.get_or_create(forge, project_id, mr_iid, job_id)
+        with transaction(commit=True) as session:
+            comment = cls()
+            comment.comment_id = comment_id
+            comment.created_at = datetime.datetime.now(datetime.timezone.utc)
+            comment.merge_request_job_id = mr_job.id
+            session.add(comment)
+            session.flush()
+            return comment.id
+
+    @classmethod
+    def get_by_id(
+        cls,
+        id_: int,
+    ) -> Optional["Comments"]:
+        """Search for a given PostgreSQL id"""
+        with transaction(commit=False) as session:
+            comment = session.query(cls).filter_by(id=id_).first()
+            return comment
+
+    @classmethod
+    def get_by_details(  # pylint: disable=too-many-arguments disable=too-many-positional-arguments
+        cls,
+        forge: Forge,
+        project_id: int,
+        mr_iid: int,
+        job_id: int,
+        comment_id: str,
+    ) -> Optional["Comments"]:
```

Comment:
If we have the `comment_id` and forge, that should be all we need to retrieve this since it's unique and indexed. I think we need two functions here: one that returns all of the comments associated with (forge, project_id, mr_iid) and a function that just returns a single Comment by the `forge_instance` and `comment_id`.